pkg/scheduler/scheduler.go

```diff
@@ -107,19 +107,20 @@ func (s *Scheduler) schedule(ctx context.Context) {
 	// 4. Sort entries based on borrowing and timestamps.
 	sort.Sort(entryOrdering(entries))
 
-	// 5. Admit entries, ensuring that no more than one workload gets
-	// admitted by a cohort (if borrowing).
-	// This is because there can be other workloads deeper in a clusterQueue whose
-	// head got admitted that should be scheduled in the cohort before the heads
-	// of other clusterQueues.
-	usedCohorts := sets.NewString()
+	// 5. Admit entries, but disallow borrowing in the cohort there are
+	// workloads that don't require borrowing in a ClusterQueue of if other
+	// ClusterQueues borrow in this cycle.
+	noBorrowingCohorts := sets.NewString()
```

Comment:
nit: Cohorts in this set may actually be borrowing already or start to borrow in this cycle. I suggest renaming it to indicate that further borrowing is suspended to reclaim the quota, maybe: `noMoreBorrowingCohorts`, or `suspendBorrowingCohorts`?